### ISSUE TITLE
Redirected to wrong url when logging in to forum.

### ISSUE THREAD

**Issue by joolswills** [NodeBB/NodeBB#5205]
* NodeBB version and git hash 1.3.0 / 4db89407c4dbe9850d925bd9529a1d106419531f
* Database (mongo or redis) and it's version. Mongodb 3.2.10 + Redis 3.2.5
* Nginx proxying to multiple nodebb processes.
* Exact steps to cause this issue
    1. Go to https://retropie.org.uk/forum/
    2. Log out/Log back in
* What you expected
    * To end up on the Forum home logged in
* What happened instead
    * redirected to https://retropie.org.uk/?loggedin instead of https://retropie.org.uk/forum/?loggedin (but am logged in if returning to forum)

**Comment by julianlam** [comment]
Seems like your `url` in `config.json` is not set correctly. It should contain the `/forum` suffix.

**Comment by joolswills** [comment]
It does - this is a new issue I believe since 1.3.x

```
{
    "url": "https://retropie.org.uk/forum",
    "bind_address": "127.0.0.1",
    "secret": "XX",
    "database": "mongo",
    "port": [4567, 4568, 4569, 4570, 4571],
    "mongo": {
        "host": "127.0.0.1",
        "port": "27017",
        "username": "XX",
        "database": "XX",
        "password": "XX"
    },
    "redis": {
        "host":"127.0.0.1",
        "port":"6379",
        "database": 0
    }
}
```

**Comment by julianlam** [comment]
Unfortunately I cannot reproduce this issue, when I log in with a subfolder configuration, I am sent back to `/forum/`.

**Comment by joolswills** [comment]
can you create an account on our forum and try? I can give you access to debug further if needed or is there anything more I can provide to help debug ?

**Comment by joolswills** [comment]
This fixes it for me - but I guess this isn't the correct fix as it seems a bit obvious (I assume data should contain the forum path ?)

```
diff --git a/public/src/client/login.js b/public/src/client/login.js
index f798347..9a73396 100644
--- a/public/src/client/login.js
+++ b/public/src/client/login.js
@@ -28,7 +28,7 @@ define('forum/login', ['translator'], function (translator) {
                                                'x-csrf-token': config.csrf_token
                                        },
                                        success: function (data, status) {
-                                               window.location.href = data + '?loggedin';
+                                               window.location.href = config.relative_path + data + '?loggedin';
                                        },
                                        error: function (data, status) {
                                                if (data.status === 403 && data.responseText === 'Forbidden') {
```

**Comment by julianlam** [comment]
Thanks for looking into the issue. Unfortunately that won't fix it as `data` in that code contains the return url. For me, it returns `/forum/` as expected (or if in a topic, returns you to `/forum/topic/123/test-topic`.

If we added that in there, it would work for you, but break for everybody else 😁

**Comment by joolswills** [comment]
I don't understand why it would be different in my case - could it be related to the nginx proxy ?

Did you trying logging out / back in a few times when you tested? I wonder what could be different in my set up.

**Comment by julianlam** [comment]
Possibly.

Do you have both `location /forum` and `proxy_pass http://ip_address:port/forum`? (with the proper values, of course)...

I find `/forum` has to be present on both.

**Comment by joolswills** [comment]
```
upstream io_nodes {
  ip_hash;
  server 127.0.0.1:4567;
  server 127.0.0.1:4568;
  server 127.0.0.1:4569;
  server 127.0.0.1:4570;
  server 127.0.0.1:4571;
}

server {
... other bits

  location ^~ /forum {
    error_page 502 =503 @maintenance;

    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header Host $http_host;
    proxy_set_header X-NginX-Proxy true;

    proxy_pass http://io_nodes/forum;
    proxy_redirect off;

    # Socket.IO Support
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
  }
```

**Comment by julianlam** [comment]
Can you get this reproduced on a development instance? I ask because if it did only start occuring in v1.3.0, then you could use `git bisect` and figure out which commit it started in...

**Comment by joolswills** [comment]
Will do. I'm not certain it did start at v1.3.0 now, but I will come back with some more information.

**Comment by stuwil** [comment]
I can reproduce without nginx in my dev instance, when using a relative path. Bug was introduced in 0590a4f2cf2390ffa628fce60520c28dc80636cd as part of the fix for #5156 - the entire URL including the relative path are removed from the x-return-to header, instead of just the base url.

**Comment by Xseba360** [comment]
I can confirm this happens to me since upgrading to 1.3.0 from 1.2.1, This breaks logging in and registering via any SSO plugins.

**Comment by julianlam** [comment]
Thanks @stuwil and @Xseba360 for your help triaging :smile:

**Comment by stuwil** [comment]
No problem! I tested the fix in #5212 against root and relative paths, using native login and the Google SSO plugin. Not sure if there are any other login scenarios I missed...
